### DISCUSSION TITLE
How do I create cumulative payout saved in unliked data

### DISCUSSION THREAD

**Discussion by slussinator** [ctreffe/alfred#240] [Alfred Q&A] [answered]
Dear Community, 

I would like to incentivize participants based on their performance in two different tasks. The payout should accumulate over both tasks. I managed to create a minimal experiment, though am unable to reproduce the problem. This is partly, because the minimal experiment does not produce an unlinked data page. 
With the minimal experiment, the calculation of the payout seems to be working, I cannot check whether it is then saved to the unlinked data page. In my actual experiment the unlinked data page exists, the payout is not saved though (neither in the actual experiment data, nor in the unlinked data). 

Can you see whether and where I am going wrong? And  could you further let me know how to add unlinked data to my minimal experiment (in order to try to reproduce the problem)?

Thanks in advance

```
import alfred3 as al
exp = al.Experiment()


@exp.member
class MAPECalc(al.AutoForwardPage):
    name = "mape_calc"
    timeout = "0s"

    def on_first_show(self):

        #adata = self.exp.adata.items()

        # Calculate individual mape
        i_mape = 100
        self += al.Value(i_mape, name=f"i_mape")


        # payout
        if i_mape <= 20:
            payout = 5
        elif i_mape <= 30:
            payout = 3
        elif i_mape <= 40:
            payout = 1
        else:
            payout = 5
        self += al.Value(payout, name="payout_i")


@exp.member
class PayoutCalc(al.AutoForwardPage):
    name = "payout_calc"
    timeout = "0s"

    def on_first_show(self):
       # adata = self.exp.adata.items()

        manual_mape = 100

        # payout
        payout = self.exp.values.get("payout_i")
        if manual_mape <= 20:
            payout += 5
        elif manual_mape <= 30:
            payout += 3
        elif manual_mape <= 40:
            payout += 1
        else:
            payout += 5

        self.exp.adata["payout"] = payout


if __name__ == "__main__":
    exp.run()

```

**Comment by jobrachem** [comment]
Hi @slussinator! Unlinked data is added with the corresponding page: [docs](https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.page.UnlinkedDataPage.html?highlight=unlinked#alfred3.page.UnlinkedDataPage)

You can use it like any other page - so maybe if you include this in your minimal example, you will be able to reproduce your issue.

**Reply by slussinator** [reply]
So I added an unlinked data page, but am now quite confused about where this happens in my original experiment. I can't find "unliked" anywhere in my script.py. Not sure how the payout values were added to the unlinked data page before... 
Also, I am unable to reproduce the problem. In my minimal exp the value is saved just fine. I guess the mistake is somewhere else...
Thanks anyways!

**Comment by jbengelsdorf** [comment, marked as answer]
First, for the community, a minimal experiment that demonstrates the desired functionality:

```python
import alfred3 as al

exp = al.Experiment()


@exp.member
class MAPE01(al.AutoForwardPage):
    name = "mape_payout_01_page"
    timeout = "0s"

    def on_first_show(self):

        # Define individual mape for task 1
        mape_01 = 100
        # Save individual mape for task 1
        self += al.Value(mape_01, name=f"mape_task01")

        # Define payout for task 1
        if mape_01 <= 50:
            payout = 2
        else:
            payout = 1
        # Save payout for task 1
        self += al.Value(payout, name="payout_task01")


@exp.member
class MAPE02(al.AutoForwardPage):
    name = "mape_payout_02_page"
    timeout = "0s"

    def on_first_show(self):

        # Get payout for task 1
        payout = self.exp.values.get("payout_task01")

        # Define individual mape for task 2
        mape_02 = 50
        # Save individual mape for task 2
        self += al.Value(mape_02, name=f"mape_task02")

        # Add payout for task 2 to payout from task 1
        if mape_02 <= 50:
            payout += 2
        else:
            payout += 1

        # Save overall payout
        self.exp.adata["payout"] = payout


@exp.member
class Payout(al.UnlinkedDataPage):
    title = "Payout Page"
    name = "payout_page"

    def on_exp_access(self):
        # Add Alert to provide information
        self += al.Alert("You can collect the necessary unlinked data here.")

    def on_close(self):
        # Add previously calculated overall payout to unlinked_data
        self += al.Value(self.exp.adata["payout"], name="payout")
        # After the payout is recorded in the unlinked_data,
        # it is overwritten in the experimental dataset so that
        # no reference can be made between unlinked and experimental data.
        self.exp.adata["payout"] = "deleted"


if __name__ == "__main__":
    exp.run()
```

I suspect that @slussinator is working with customized classes such as the `RegistrationPage`. This is a class derived from the `UnlinkedDataPage`, which is defined outside the script.py and does not belong to the standard classes of alfred3. This would explain why no "unlinked" can be found in the script.py. This particular customized class requires (a) the final payout to be stored as `self.exp.adata["payout"]` (this can be set in the script.py) and (b) `option_1_register_payout=True` must be set in the associated dictionary for the respective payout option (this can be set in f. e. the content.py).

For better understanding, here is a folder containing the `RegistrationPage` code (in the exp_registration.py file):
[2.0.0_ExpRegistration.zip](https://github.com/ctreffe/alfred/files/11464628/2.0.0_ExpRegistration.zip)

**Reply by slussinator** [reply]
Thanks for the help :) The problem was, that the "option_X_register_payout=" was set to "False".
